applications/asset_tracker_v2/src/modules/data_module.c

```diff
@@ -681,6 +697,117 @@ static void requested_data_list_set(enum app_module_data_type *data_list,
+
+		if (current_cfg.active_mode) {
+			LOG_WRN("New Device mode: Active");
+		} else {
+			LOG_WRN("New Device mode: Passive");
+		}
+
+		config_change = true;
+		current_cfg.active_mode_valid = true;
+	}
+
+	if (new_config->gps_timeout > 0) {
+		if (current_cfg.gps_timeout != new_config->gps_timeout) {
+			current_cfg.gps_timeout = new_config->gps_timeout;
+
+			LOG_WRN("New GPS timeout: %d", current_cfg.gps_timeout);
+
+			config_change = true;
+			current_cfg.gps_timeout_valid = true;
+		}
+	} else {
+		LOG_ERR("New GPS timeout out of range: %d", new_config->gps_timeout);
+	}
+
+	if (new_config->active_wait_timeout > 0) {
+		if (current_cfg.active_wait_timeout != new_config->active_wait_timeout) {
+			current_cfg.active_wait_timeout = new_config->active_wait_timeout;
+
+			LOG_WRN("New Active wait timeout: %d", current_cfg.active_wait_timeout);
+
+			config_change = true;
+			current_cfg.active_wait_timeout_valid = true;
+		}
+	} else {
+		LOG_ERR("New Active timeout out of range: %d", new_config->active_wait_timeout);
+	}
+
+	if (new_config->movement_resolution > 0) {
+		if (current_cfg.movement_resolution != new_config->movement_resolution) {
+			current_cfg.movement_resolution = new_config->movement_resolution;
+
+			LOG_WRN("New Movement resolution: %d", current_cfg.movement_resolution);
+
+			config_change = true;
+			current_cfg.movement_resolution_valid = true;
+		}
+	} else {
+		LOG_ERR("New Movement resolution out of range: %d",
+			new_config->movement_resolution);
+	}
+
+	if (new_config->movement_timeout > 0) {
+		if (current_cfg.movement_timeout != new_config->movement_timeout) {
+			current_cfg.movement_timeout = new_config->movement_timeout;
+
+			LOG_WRN("New Movement timeout: %d", current_cfg.movement_timeout);
+
+			config_change = true;
+			current_cfg.movement_timeout_valid = true;
+		}
+	} else {
+		LOG_ERR("New Movement timeout out of range: %d", new_config->movement_timeout);
+	}
+
+	if ((new_config->accelerometer_threshold < ACCELEROMETER_S_M2_MAX) &&
+	    (new_config->accelerometer_threshold > 0)) {
+		if (current_cfg.accelerometer_threshold != new_config->accelerometer_threshold) {
+			current_cfg.accelerometer_threshold = new_config->accelerometer_threshold;
+
+			LOG_WRN("New Accelerometer threshold: %f",
+				current_cfg.accelerometer_threshold);
+
+			config_change = true;
+			current_cfg.accelerometer_threshold_valid = true;
+		}
+	} else {
+		LOG_ERR("New Accelerometer threshold out of range: %f",
+			new_config->accelerometer_threshold);
+	}
+
+	err = save_config(&current_cfg, sizeof(current_cfg));
+	if (err) {
+		LOG_WRN("Configuration not stored, error: %d", err);
+	}
+
+	/* Distribute the configuration to other modules regardless
+	 * if the values has changed or not. This is to make sure that
+	 * the cloud module (which does the initial decoding of the
+	 * incoming configuration) always has the latest valid
+	 * configuration.
+	 */
+	config_distribute(DATA_EVT_CONFIG_READY);
```

Comment:
Could include a flag, whether the configuration has actually change. So other modules can ignore this event, if the configuration is left unchange.

Reply:
Noticed some potential fishy business when looking at this. Will address it in a future PR.